health/tracker.go

```diff
@@ -750,6 +751,23 @@ func (t *Tracker) SetLatestBlockNumber(upstream common.Upstream, blockNumber int
 		g := t.getLatestBlockGauge(t.projectId, "*", netLabel, "*")
 		g.Set(float64(blockNumber))
 		needsGlobalUpdate = true
+
+		// Atomically update timestamp when network-level block number is updated
+		if blockTimestamp > 0 {
+			oldTimestamp := ntwMeta.evmLatestBlockTimestamp.Load()
+			if blockTimestamp > oldTimestamp {
+				ntwMeta.evmLatestBlockTimestamp.Store(blockTimestamp)
+
+				// Calculate and record distance metric from EVM state poller
+				currentTime := time.Now().Unix()
+				distance := currentTime - blockTimestamp
+				telemetry.MetricNetworkLatestBlockTimestampDistance.WithLabelValues(
+					t.projectId,
+					netLabel,
+					"evm_state_poller",
+				).Set(float64(distance))
```

Comment:
Will negative values work correctly here? I assume in cases where a node is wrongly in future it should correctly show negative to proper inform the user